### DISCUSSION TITLE
icon not centering itself in Input

### DISCUSSION THREAD

**Discussion by somguyth** [chakra-ui/chakra-ui#4402] [Q&A] [answered]
I'm trying to position an icon correctly in the middle of an input, where that input has `size="xs"`. However, all my attempts have failed, and the icon keeps being positioned too low/down. I've also tried using `IconButton` instead of a regular icon but that didn't work either.

![image](https://user-images.githubusercontent.com/37128022/126064791-72119847-8adf-41e6-b962-56ab2c6939fe.png)

```
import { Input, InputGroup, InputLeftElement } from "@chakra-ui/react"
import { SearchIcon } from "@chakra-ui/icons"

// ...

    <InputGroup>
      <InputLeftElement
        pointerEvents="none"
        children={<SearchIcon color="gray.300"/>}
        size="xs"
        />
      <Input
        variant="outline"
        placeholder={``}
        size="xs"
      />
    </InputGroup>
```

What am I doing wrong?

Here's the codesandbox. Note that in this codesandbox, the icon is actually above the middle of Input (which is still wrong), instead of below as on my local machine.

https://codesandbox.io/s/optimistic-bartik-5ifsd?file=/src/App.tsx

**Comment by payapula** [comment, marked as answer]
@somguyth  I have tried your [codesandbox](https://codesandbox.io/s/optimistic-bartik-5ifsd?file=/src/App.tsx) reproduction, you forgot to add `ChakraProvider`. Having said that, Yes, on `Input` with `size="xs"` the `InputLeftElement` is not adjusted and icon seems to be fixed in its original position.

![image](https://user-images.githubusercontent.com/7134153/126066464-a7554305-86ba-4cda-8e32-f56d03c6dc32.png)

However, you can fix it by giving the prop `top="-2"` to `InputLeftElement`.

![image](https://user-images.githubusercontent.com/7134153/126066550-150cd4ff-d02f-4ffd-80b6-8853c3ad2c87.png)

Here is the fix - https://codesandbox.io/s/reverent-leftpad-pzdpu?file=/src/App.tsx

I am not sure if Chakra team needs to fix it though.

**Reply by somguyth** [reply]
This does look like a bug, no? The alignment should be automatic

**Reply by payapula** [reply]
Yes. It should be a defect.

**Reply by primos63** [reply]
To center the icon use `<InputGroup size='xs'>`. You don't need a size prop on the children.
```javascript
export default function App() {
  return (
    <>
      <div>123</div>
      <InputGroup size='md'>
        <InputLeftElement
          pointerEvents="none"
          children={<SearchIcon color="gray.300" />}
        />
        <Input variant="outline" placeholder={``} />
      </InputGroup>
    </>
  );
}
```
